src/main/java/io/debezium/connector/spanner/db/metadata/SchemaRegistry.java

```diff
@@ -89,38 +94,70 @@ private boolean validate(TableId tableId, List<Column> rowType) {
         return SchemaValidator.validate(schemaTable, watchedTable, rowType);
     }
 
-    public synchronized boolean updateSchema(Timestamp updatedTimestamp) {
+    public synchronized boolean updateSchema(TableId tableId, Timestamp updatedTimestamp, List<Column> rowType) {
         if (updatedTimestamp.equals(this.timestamp)) {
             return false;
         }
 
         LOGGER.info("Schema is outdated. Try to update schema registry...");
-        forceUpdateSchema(updatedTimestamp);
+        forceUpdateSchema(tableId, updatedTimestamp, rowType);
         LOGGER.info("Schema registry has been updated to date {}", updatedTimestamp);
         return true;
     }
 
-    private void forceUpdateSchema(Timestamp updatedTimestamp) {
-        this.timestamp = updatedTimestamp;
-        this.changeStream = schemaDao.getStream(timestamp, streamName);
+    // Used for unit testing only to set the mock SpannerSchema class
+    @VisibleForTesting
+    void setSpannerSchema(SpannerSchema spannerSchema) {
```

Comment:
Why do you need to set the mock SpannerSchema class? I don't think I see the other unittests needing to set this?